src/robot/libraries/String.py

```diff
@@ -403,13 +441,25 @@ def get_lines_matching_regexp(
         match = regexp.search if partial_match else regexp.fullmatch
         return self._get_matching_lines(string, match)
 
-    def _get_matching_lines(self, string, matches):
+    # TODO(@pekkaklarck): some paramters could be string, some bytes -> risk of incompatibilies
+    def _get_matching_lines(
+        self,
+        string: str,
+        matches: Callable[[str], "bool | re.Match[str] | None"],
+    ) -> str:
         lines = string.splitlines()
         matching = [line for line in lines if matches(line)]
         logger.info(f"{len(matching)} out of {len(lines)} lines matched.")
         return "\n".join(matching)
 
-    def get_regexp_matches(self, string, pattern, *groups, flags=None):
+    # TODO(@pekkaklarck): some paramters could be string, some bytes -> risk of incompatibilies
+    def get_regexp_matches(
+        self,
+        string: str,
+        pattern: str,
+        *groups: object,
+        flags: Flags = None,
+    ) -> list["str | tuple"]:
```

Comment:
I guess the correct typing for the return value would be `"list[str | tuple[str, ...]]"`. Notice that quotes need to be around the whole construct for Python 3.8 compatibility.